napari/layers/base/base.py

```diff
@@ -1535,7 +1536,7 @@ def _update_draw(
             ):
                 self._data_level = level
                 self.corner_pixels = corners
-                self.refresh()
+                self.events.reslice(Event('reslice', layer=self))
```

Comment:
Draws only make sense in the context of a layer being hooked up to the full viewer with vispy, so we can replace refresh with reslice here to notify the viewer model that it should slice this layer.

Reply:
We could consider replacing other use of `refresh` with `reslice` in this PR, but those represent slightly larger changes in behavior that would also require a lot of updates to tests (i.e. tests that depend on `Layer.refresh` having some side effect on the layer state).